infra/Pos/Communication/Relay/Class.hs

```diff
@@ -57,23 +58,26 @@ data MempoolParams m where
       ) => Proxy tag -> m [key] -> MempoolParams m
 
 data InvReqDataParams key contents m = InvReqDataParams
-    { contentsToKey :: contents -> m key
+    { invReqMsgType :: !MsgType
```

Comment:
> Can we do block propagation/relaying using the InvReqDataParams/DataParams/PropagationMsg style?

I think we cannot, but that's ok, it's just another message constructor in the outbound queue message type.